### PR TITLE
pandas v2.0.0

### PR DESCRIPTION
**Pull request by datapythonista**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by datapythonista** [comment]
Added `tzdata`, @MarcoGorelli do you mind having a look?

**Comment by MarcoGorelli** [comment]
from https://github.com/pandas-dev/pandas/pull/51247#issuecomment-1463124830 , it looks like you've done exactly that  - @lithomas1 do you know what the issue might be? 

the logs show
```
2023-04-03T14:01:12.6088211Z pandas 2.0.0 requires tzdata, which is not installed.
2023-04-03T14:01:12.8512388Z Tests failed for pandas-2.0.0-py38hdc8b05c_0.conda - moving package to /home/conda/feedstock_root/build_artifacts/broken
2023-04-03T14:01:12.8513965Z WARNING:conda_build.build:Tests failed for pandas-2.0.0-py38hdc8b05c_0.conda - moving package to /home/conda/feedstock_root/build_artifacts/broken
2023-04-03T14:01:12.8809618Z TESTS FAILED: pandas-2.0.0-py38hdc8b05c_0.conda
2023-04-03T14:01:19.3131987Z ##[error]Bash exited with code '1'.
```

**Comment by MarcoGorelli** [comment]
should `tzdata` be removed from `run`, and `- {{ PYTHON }} -m pip install tzdata>=2022.1` be added to `script`?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,11 +30,14 @@ requirements:
...
     - cython >=0.29.24,<3
     - numpy
+    - versioneer
+    - tomli  # [py<311]
   run:
     - python
     - {{ pin_compatible('numpy') }}
     - python-dateutil >=2.8.1
     - pytz >=2020.1
+    - tzdata >=2022a
```
Comment (ocefpaf) [inline recipe/meta.yaml:40]:
```suggestion
    - python-tzdata >=2022a
```

**Comment by alippai** [comment]
Shouldn’t the tzdata required only when using python <=3.8? Or pandas is not compatible with the zoneinfo only?

**Comment by MarcoGorelli** [comment]
I presume you mean 3.9? Pandas isn't compatible with just `ZoneInfo` yet, but once it drops 3.8 support, then it can drop `pytz` and just use `ZoneInfo`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -30,11 +30,14 @@ requirements:
...
     - cython >=0.29.24,<3
     - numpy
+    - versioneer
+    - tomli  # [py<311]
   run:
     - python
     - {{ pin_compatible('numpy') }}
     - python-dateutil >=2.8.1
     - pytz >=2020.1
+    - python-tzdata >=2022a
```
Comment (ocefpaf) [inline recipe/meta.yaml:40]: BTW, tzdata is included in modern Python but pip is designed to find it only if installed by the python alternative like this. We could probably drop the `pip check` test too. But "overi-nstalling" tzdata is not a big deal and probably better than dropping the pip-check.

**Comment by datapythonista** [comment]
Seems like we have a failure in a build:

```
Traceback (most recent call last):
  File "C:\Miniforge\Scripts\conda-mambabuild-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 256, in main
    call_conda_build(action, config)
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 228, in call_conda_build
    result = api.build(
  File "C:\Miniforge\lib\site-packages\conda_build\api.py", line 180, in build
    return build_tree(
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 3089, in build_tree
    test(pkg, config=metadata.config.copy(), stats=stats)
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 2879, in test
    environ.create_env(metadata.config.test_prefix, actions, config=metadata.config,
  File "C:\Miniforge\lib\site-packages\conda_build\environ.py", line 1052, in create_env
    display_actions(actions, index)
  File "C:\Miniforge\lib\site-packages\conda\exports.py", line 204, in display_actions
    actions['LINK'] = [index[d] for d in actions['LINK']]
  File "C:\Miniforge\lib\site-packages\conda\exports.py", line 204, in <listcomp>
    actions['LINK'] = [index[d] for d in actions['LINK']]
KeyError: Dist(channel='conda-forge', dist_name='pytz-2023.3-pyhd8ed1ab_0', name='pytz', fmt='.conda', version='2023.3', build_string='pyhd8ed1ab_0', build_number=0, base_url=None, platform=None)
```

I guess something went wrong with conda. @ocefpaf have you seen this before? Do you think we should just rerun that build?

**Comment by ocefpaf** [comment]
> I guess something went wrong with conda. @ocefpaf have you seen this before? Do you think we should just rerun that build?

It may requite a restart. Let's wait for the rest of the CIs to finish before we restart it.

**Comment by ocefpaf** [comment]
BTW, we should wait for https://github.com/conda-forge/status/issues/143 to be resolved before merging this one.

**Comment by ocefpaf** [comment]
The ppc ones had some real failures. Looks like precision stuff and maybe be same to skip, not sure.

**Comment by datapythonista** [comment]
Thanks for the help and for the info.

We found the error with ppc on the release candidates, we're happy to release without ppc. Hopefully we have a fix for pandas 2.0.1. Let's see if we're more lucky with the second attempt of the other error.

It starts to be late in Dubai where I am, but for anyone else in a better time zone, feel free to merge this if only ppc is failing. Otherwise I'll take care of it tomorrow morning my time.

**Pull request merged**